src/app/shared/components/multi-select/multi-select.component.ts

```diff
@@ -0,0 +1,134 @@
+  isSelected: boolean
+}
+
+@Component({
+  selector: 'sofie-multi-select',
+  templateUrl: './multi-select.component.html',
+  styleUrls: ['./multi-select.component.scss'],
+  providers: [
+    {
+      provide: NG_VALUE_ACCESSOR,
+      multi: true,
+      useExisting: MultiSelectComponent,
+    },
+  ],
+})
+export class MultiSelectComponent<T> implements OnInit, ControlValueAccessor {
+  protected Icon = Icon
+  protected IconSize = IconSize
+
+  @Input() public options: MultiSelectOption<T>[] = []
+  public selectableOptions: SelectableOption<T>[] = []
+
+  @Input() public label: string
+  @Input() public placeholder?: string
+
+  @Output() public onChange: EventEmitter<T[]> = new EventEmitter()
+
+  public selectedOptions: SelectableOption<T>[] = []
+
+  private onChangeCallback: (value: T[]) => void
+  private onTouchedCallback: () => void
+
+  private values: T[] = []
+
+  private isTouched: boolean = false
+
+  constructor() {}
+
+  public ngOnInit(): void {
+    this.selectableOptions = this.options.map(option => {
+      return {
+        ...option,
+        isSelected: false,
+      }
+    })
+  }
+
+  public toggleOption(option: SelectableOption<T>): void {
+    this.markAsTouched()
+    option.isSelected = !option.isSelected
+    if (option.isSelected) {
+      this.selectedOptions.push(option)
+      this.updateValuesFromSelectedOptions()
+      return
+    }
+    const index: number = this.selectedOptions.indexOf(option)
+    if (index < 0) {
+      return
+    }
+    this.selectedOptions.splice(index, 1)
+
+    this.updateValuesFromSelectedOptions()
+  }
+
+  private markAsTouched(): void {
+    if (this.isTouched || !this.onTouchedCallback) {
+      return
+    }
+    this.onTouchedCallback()
+    this.isTouched = true
+  }
+
+  private updateValuesFromSelectedOptions(): void {
+    this.values = this.selectedOptions.map(option => option.value)
+    this.onChange.emit(this.values)
+    if (!this.onChangeCallback) {
+      return
+    }
+    this.onChangeCallback(this.values)
+  }
+
+  public getSelectedOptionsAsDisplay(): string {
+    return this.selectedOptions
+      .map(option => option.name)
+      .reduce((a, b) => {
+        return `${a}, ${b}`
+      })
+  }
+
+  public clearAllSelected(): void {
+    this.clearSelectedOptions()
+    this.updateValuesFromSelectedOptions()
+  }
+
+  private clearSelectedOptions(): void {
+    this.selectableOptions.map(option => (option.isSelected = false))
+    this.selectedOptions = []
+  }
+
+  public writeValue(value: T[]): void {
+    this.values = value
+    this.updateSelectedOptionsFromValues()
+  }
+
+  private updateSelectedOptionsFromValues(): void {
+    this.clearSelectedOptions()
+    this.values.forEach(value => {
+      const option: SelectableOption<T> | undefined = this.selectableOptions.find(option => option.value === value)
+      if (!option) {
+        return
+      }
+      option.isSelected = true
+      this.selectedOptions.push(option)
+    })
```

Comment:
Updated. However I did not create a new object (`{...option, isSelected: true}`) because this loses the object reference and then the selected options won't have a checkmark next to them in the dropdown menu.